lib/runtime/procs/base.js

```diff
@@ -32,6 +33,16 @@ var base = Base.extend({
         this.logger = JuttleLogger.getLogger(this.logger_name);
     },
 
+    validate_options: function(allowedOptions) {
+        var unknown = _.difference(_.keys(this.options), allowedOptions);
+        if (unknown.length > 0) {
+            throw new this.compile_error('RT-UNKNOWN-OPTION-ERROR', {
```

Comment:
`this.compile_error` will always attach the location of the proc to the error so it's useful.

I actually think we should add a `this.compile_error` (or `this.compileError`) function to the adapters in order to make the calling conventions consistent.

Also note that in neither case should there be an explicit `new` -- the function returns a new instance so the proper usage is `throw this.compile_error(...)`.